jabgui/src/main/java/org/jabref/gui/util/SelectableTextFlow.java

```diff
@@ -0,0 +1,157 @@
+package org.jabref.gui.util;
+
+import java.util.Optional;
+
+import javafx.geometry.Point2D;
+import javafx.scene.Cursor;
+import javafx.scene.Node;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.layout.Pane;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Path;
+import javafx.scene.shape.PathElement;
+import javafx.scene.text.HitInfo;
+import javafx.scene.text.Text;
+import javafx.scene.text.TextFlow;
+
+import org.jabref.gui.ClipBoardManager;
+
+import com.airhacks.afterburner.injection.Injector;
+
+/**
+ * A TextFlow that allows text selection and copying.
+ */
+public class SelectableTextFlow extends TextFlow {
+    private Optional<HitInfo> startHit = Optional.empty();
+    private Optional<HitInfo> endHit = Optional.empty();
+    private Optional<Path> selectionPath = Optional.empty();
+
+    private final Pane parentPane;
+
+    /**
+     * Creates a SelectableTextFlow instance.
+     *
+     * @param parent The parent Pane to which this TextFlow will be added.
+     */
+    public SelectableTextFlow(Pane parent) {
+        this.parentPane = parent;
+        setCursor(Cursor.TEXT);
+        setFocusTraversable(true);
+
+        setOnMouseClicked(_ -> {
+            requestFocus();
+            deselect();
+        });
+
+        setOnMousePressed(event -> {
+            requestFocus();
+            handleMousePressed(event);
+        });
+
+        setOnMouseDragged(event -> {
+            requestFocus();
+            handleMouseDragged(event);
+        });
+
+        focusedProperty().addListener((_, _, newFocus) -> {
+            if (!newFocus) {
+                deselect();
+            }
+        });
+    }
+
+    /**
+     * Copy selected text.
+     */
+    public void copySelectedText() {
+        if (startHit.isEmpty() || endHit.isEmpty()) {
+            return;
+        }
+
+        int startIndex = Math.min(startHit.get().getCharIndex(), endHit.get().getCharIndex());
+        int endIndex = Math.max(startHit.get().getCharIndex() + 1, endHit.get().getCharIndex() + 1);
+
+        String fullText = getTextFlowContent();
+        if (startIndex >= 0 && endIndex <= fullText.length() && startIndex < endIndex) {
+            String selectedText = fullText.substring(startIndex, endIndex);
+            ClipBoardManager clipBoardManager = Injector.instantiateModelOrService(ClipBoardManager.class);
+            clipBoardManager.setContent(selectedText);
+        }
+    }
+
+    /**
+     * Select all text in the TextFlow.
+     */
+    public void selectAll() {
+        if (getChildren().isEmpty()) {
+            return;
+        }
+        startHit = Optional.of(hitTest(new Point2D(0, 0)));
+        endHit = Optional.of(hitTest(new Point2D(getLayoutBounds().getWidth(), getLayoutBounds().getHeight())));
+        updateSelectionHighlight();
+    }
+
+    /**
+     * Clears selection.
```

Comment:
Maybe rename the method then - and remove the JavaDoc as the method name already contains it.